src/librustc_target/spec/nvptx64_nvidia_cuda.rs

```diff
@@ -0,0 +1,73 @@
+use spec::{LinkerFlavor, Target, TargetOptions, TargetResult, PanicStrategy, MergeFunctions};
+use spec::abi::Abi;
+
+pub fn target() -> TargetResult {
+    Ok(Target {
+        arch: "nvptx64".to_string(),
+        data_layout: "e-i64:64-i128:128-v16:16-v32:32-n16:32:64".to_string(),
+        llvm_target: "nvptx64-nvidia-cuda".to_string(),
+
+        target_os: "cuda".to_string(),
+        target_vendor: "nvidia".to_string(),
+        target_env: String::new(),
+
+        linker_flavor: LinkerFlavor::PtxLinker,
+
+        target_endian: "little".to_string(),
+        target_pointer_width: "64".to_string(),
+        target_c_int_width: "32".to_string(),
+
+        options: TargetOptions {
+            // The linker can be installed from `crates.io`.
+            linker: Some("rust-ptx-linker".to_string()),
+
+            // With `ptx-linker` approach, it can be later overriden via link flags.
+            cpu: "sm_20".to_string(),
```

Comment:
Recent CUDA versions (at least since 10.0) have dropped support for compute capabilities less than 3.x. In other words `ptxas` won't compile this. I would recommend changing the minimum target-cpu to "sm_30".

Reply:
Hmmm, actually `ptxas` _will_ compile this as long as you don't specify args `-arch sm_20`, but then setting "sm_20" in the PTX may not be doing what we think it is doing if `ptxas` is emitting code for at least "sm_30" anyway.